crc/services/temp_migration_service.py

```diff
@@ -0,0 +1,262 @@
+# from crc import app, session
+from crc.models.file import FileModel, FileModelSchema, FileDataModel, CONTENT_TYPES
+from crc.models.workflow import WorkflowSpecModel, WorkflowSpecModelSchema, WorkflowSpecCategoryModel, WorkflowSpecCategoryModelSchema
+from crc.services.file_service import FileService
+from crc.services.spec_file_service import SpecFileService
+from crc.services.workflow_service import WorkflowService
+from sqlalchemy import desc
+
+import json
+import os
+
+
+# SYNC_FILE_ROOT = SpecFileService.get_sync_file_root()
+
+
+class FromFilesystemService(object):
+
+    @staticmethod
+    def process_directory(directory):
+        files = []
+        directories = []
+        directory_items = os.scandir(directory)
+        for item in directory_items:
+            if item.is_dir():
+                directories.append(item)
+            elif item.is_file():
+                files.append(item)
+
+        return files, directories
+
+    @staticmethod
+    def process_workflow_spec(json_file, directory):
+        file_path = os.path.join(directory, json_file)
+
+        with open(file_path, 'r') as f_open:
+            data = f_open.read()
+            data_obj = json.loads(data)
+            workflow_spec_model = session.query(WorkflowSpecModel).\
+                filter(WorkflowSpecModel.id == data_obj['id']).\
+                first()
+            if not workflow_spec_model:
+                category_id = None
+                if data_obj['category'] is not None:
+                    category_id = session.query(WorkflowSpecCategoryModel.id).filter(
+                        WorkflowSpecCategoryModel.display_name == data_obj['category']['display_name']).scalar()
+                workflow_spec_model = WorkflowSpecModel(id=data_obj['id'],
+                                                        display_name=data_obj['display_name'],
+                                                        description=data_obj['description'],
+                                                        is_master_spec=data_obj['is_master_spec'],
+                                                        category_id=category_id,
+                                                        display_order=data_obj['display_order'],
+                                                        standalone=data_obj['standalone'],
+                                                        library=data_obj['library'])
+                session.add(workflow_spec_model)
+                session.commit()
+
```

Comment:
Should likely be an 'else' here that would update the model, which is also possible with marshmallow.